x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/xcontent/XContentUtils.java

```diff
@@ -94,16 +98,22 @@ public static void addAuthorizationInfo(final XContentBuilder builder, final Map
         if (authKey == null) {
             return;
         }
+        Subject authenticationSubject;
+        try {
+            authenticationSubject = AuthenticationContextSerializer.decode(authKey).getEffectiveSubject();
+        } catch (Exception e) {
+            logger.warn("Failed to decode auth key while adding auth info to REST response", e);
```

Comment:
Can we log the `authKey` string as well which is useful for diagnosis? It can be at the debug level if you have concerns.

Reply:
Nevermind. I think it is better for us to log it inside `AuthenticationContextSerializer.decode`.

Reply:
OK, I moved the logging into `AuthenticationContextSerializer` and also log the string being decoded now.